### PR TITLE
simple reorg changes and new detailed change types

### PR DESCRIPTION
**Pull request by evykassirer**

### CODE REVIEW

File: lib/MathChangeTypes.js [hunk 1]
```diff
@@ -7,26 +7,28 @@ module.exports = {
...
   ADD_COEFFICIENT_OF_ONE: 'No coefficient is the same a coefficient equal to 1',
   ADD_EXPONENT_OF_ONE: 'No exponent is the same as a coefficient of 1',
   GROUP_COEFFICIENTS: 'Group coefficients together',
   COLLECT_EXPONENTS: 'Sum up the exponents',
   MULTIPLY_COEFFICIENTS: 'Multiply coefficients together',
   ADD_POLYNOMIAL_TERMS: 'Add terms together',
-  MULT_POLYNOMIAL_TERMS: 'Multiply terms together',
+  MULTIPLY_POLYNOMIAL_TERMS: 'Multiply terms together',
   REARRANGE_COEFF: 'Move the coefficient in front',
+  REDUCE_EXPONENT_BY_ZERO: 'reduce-exponent-by-zero',
```
Comment (evykassirer) [inline lib/MathChangeTypes.js]: these actually need real descriptions until Ahmed's change goes in, cause we'll be deploying it to prod probably

File: lib/evaluateArithmetic.js [hunk 2]
```diff
@@ -13,9 +13,15 @@ function evaluateArithmeticDFS(node) {
   let innerNodeStatus;
   if (NodeType.isParenthesis(node)) {
     innerNodeStatus = evaluateArithmeticDFS(node.content);
+    if (innerNodeStatus.hasChanged()) {
```
Comment (aelnaiem) [inline lib/evaluateArithmetic.js]: Can we just call this status? that's what you call it in other places. Or should it always be innerNodeStatus?

File: lib/simplifyBasics.js [hunk 3]
```diff
@@ -99,6 +90,74 @@ function simplifyBasics(node) {
...
+      MathChangeTypes.MULTIPLY_BY_ZERO, oldNode, newNode);
+  }
+  else {
+    return NodeStatus.noChange(node);
+  }
+}
+
+// If `node` is a fraction with 0 as the numerator, reduce the node to 0.
+// Returns a NodeStatus object.
+function reduceZeroDivdedByAnything(node) {
```
Comment (aelnaiem) [inline lib/simplifyBasics.js]: Divded*

File: lib/evaluateArithmetic.js (hunk 2 above)
Comment (evykassirer) [inline lib/evaluateArithmetic.js]: yeah that's a good question

which did you find easier to understand from a "less familiar with the code" viewpoint when you looked over the code last week?

**Pull request merged**